In [40]:
import pandas as pd
import numpy as np

In [41]:
data_original = pd.read_csv('C:/Users/yakan/OneDrive/Documents/Thinkful/DS_Bootcamp/Unit1_data_and_analysis_ds_for_investigations/Lesson4_experimental_design/WELLCOME_APCspend2013_forThinkful.csv', 
                            encoding = 'latin-1')

In [42]:
data_original.head()

PMID/PMCID Publisher           Journal title  \
0                    NaN       CUP  Psychological Medicine   
1             PMC3679557       ACS       Biomacromolecules   
2  23043264  PMC3506128        ACS              J Med Chem   
3    23438330 PMC3646402       ACS              J Med Chem   
4   23438216 PMC3601604        ACS              J Org Chem   

                                       Article title  \
0  Reduced parahippocampal cortical thickness in ...   
1  Structural characterization of a Model Gram-ne...   
2  Fumaroylamino-4,5-epoxymorphinans and related ...   
3  Orvinols with mixed kappa/mu opioid receptor a...   
4  Regioselective opening of myo-inositol orthoes...   

  COST (£) charged to Wellcome (inc VAT when charged)  
0                                              £0.00   
1                                           £2381.04   
2                                            £642.56   
3                                            £669.64   
4                                            £685.88

### Process data to resolve string issues.

In [43]:
#upcase everything in journal title column
data_original['j_title'] = data_original['Journal title'].str.upper()

In [44]:
#get rid of leading and trailing space in the string journal title column
data_original['j_title'] = data_original['j_title'].str.strip() 

In [45]:
#aggregate up by journal title
#this gets a count, so it becomes obvious if there are a large count of the same
#journal title that is misspelled.
data_original['j_title'].value_counts()

PLOS ONE                                                              190
JOURNAL OF BIOLOGICAL CHEMISTRY                                        53
NEUROIMAGE                                                             29
NUCLEIC ACIDS RESEARCH                                                 26
PLOS PATHOGENS                                                         24
PLOS GENETICS                                                          24
PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES                        22
PLOS NEGLECTED TROPICAL DISEASES                                       20
HUMAN MOLECULAR GENETICS                                               19
NATURE COMMUNICATIONS                                                  19
MOVEMENT DISORDERS                                                     15
BMC PUBLIC HEALTH                                                      14
BRAIN                                                                  14
JOURNAL OF NEUROSCIENCE               

In [46]:
#start reasigning misspelled or incorrect journal title names.
data_original['j_title'] = data_original['j_title'].replace({'NATIONAL ACADEMY OF SCIENCES': 'PNAS',
                                                    'PROCEEDINGS OF THE NATIONAL ACADEMY OF SCIENCES': 'PNAS',
                                                    'PLOSONE': 'PLOS ONE',
                                                    'BMJ OPEN': 'BMJ'})

In [47]:
#these are the counts based on cleaned up grouping
data_original['j_title'].value_counts()

PLOS ONE                                                              199
JOURNAL OF BIOLOGICAL CHEMISTRY                                        53
PNAS                                                                   31
NEUROIMAGE                                                             29
NUCLEIC ACIDS RESEARCH                                                 26
PLOS PATHOGENS                                                         24
PLOS GENETICS                                                          24
PLOS NEGLECTED TROPICAL DISEASES                                       20
BMJ                                                                    19
HUMAN MOLECULAR GENETICS                                               19
NATURE COMMUNICATIONS                                                  19
MOVEMENT DISORDERS                                                     15
BRAIN                                                                  14
BMC PUBLIC HEALTH                     

In [48]:
#find other instances of titles that may be mismatched
#it seems like the most common journal, PLOS, has other types of journals that are 
#subspecialites. These need to stay separate.
data_original[(data_original['j_title'].str.contains('PLOS', 
                                                     case = False, 
                                                     regex = False)) == True]

PMID/PMCID                         Publisher  \
1278                         PMC3744396                              PLOS   
1279                         PMC3681601                              PLoS   
1280                            3715439                              Plos   
1281                         PMC3493395                              PLOS   
1282                            3517619                              PLoS   
1283                            3498109                              PLoS   
1284                            3515553                              PLoS   
1285                            3522679                              PLoS   
1286                            3485223                              PLoS   
1287                         PMC3547059                              PLoS   
1288                            3573029                              PLoS   
1289                            3769269                              PLoS   
1290                            3782430                              PLoS   
1291                            3797097                              PLoS   
1292                            3675909                              PLoS   
1293                         PMC3566186                              PLOS   
1294                         PMC3694955                              PLoS   
1295                         PMC3729655                              PLOS   
1296                         PMC3774797                              PLoS   
1297                            3620162                              PLoS   
1298                         PMC3694164                              PLOS   
1299                            3669132                              Plos   
1300                         PMC3597521                              PLOS   
1301                         PMC3798605                              PLOS   
1302                  PMCID:PMC3715547)  PLoS (Public Library of Science)   
1303                   PMCID:PMC3711905  PLoS (Public Library of Science)   
1304                  PMCID: PMC3661540  PLoS (Public Library of Science)   
1305                   PMCID:PMC3706442  PLoS (Public Library of Science)   
1306                         PMC3527205    PLoS Public Library of Science   
1345                           23239765         Public Library of Science   
...                                 ...                               ...   
1584                           23326235         Public Library of Science   
1585                           21483485         Public Library of Science   
1586                           22241985         Public Library of Science   
1587             PMCID:\n    PMC3578823         Public Library of Science   
1588             PMCID:\n    PMC3764221         Public Library of Science   
1589             PMCID:\n    PMC3192839         Public Library of Science   
1590  PMID: 23468629 /PMCID: PMC3585132         Public Library of Science   
1591                           24098129         Public Library of Science   
1592                  PMCID: PMC3789764         Public Library of Science   
1593                         PMC3656102         Public Library of Science   
1594             PMCID:\n    PMC3731244         Public Library of Science   
1595                           23505371         Public Library of Science   
1596                         PMC3585151         Public Library of Science   
1597                  PMCID: PMC3749955         Public Library of Science   
1598           PMCID:\n    PMC3764210\n         Public Library of Science   
1599                           23308065         Public Library of Science   
1600                      PMID:23633946         Public Library of Science   
1601                         PMC3610638         Public Library of Science   
1602           PMCID:\n    PMC3784482\n         Public Library of Science   
1603           PMCID:\n    PMC3605240\n         Public Library of Science   
1604             PMCID:\n

##### Seems like things are cleaned up pretty well.
### The top five journals are:
* Journal title (with number of articles)                                            
* PLOS ONE                                                   (199)
* JOURNAL OF BIOLOGICAL CHEMISTRY                             (53)
* PNAS                                                        (31)
* NEUROIMAGE                                                  (29)
* NUCLEIC ACIDS RESEARCH                                      (26)

### Rework the cost column, so that mean, median, and std dev can be computed.

In [49]:
#rename cost column to a shorter name
data_original = data_original.rename(columns = {'COST (£) charged to Wellcome (inc VAT when charged)': 'cost'})

In [57]:
#remove leading and lagging space
data_original['cost'] = data_original['cost'].str.strip()

In [69]:
#replace the non-numeric characters with nothing, so essentially remove them
data_original['cost'] = data_original['cost'].map(lambda x: x.replace('£', ''))

In [70]:
data_original['cost'] = data_original['cost'].map(lambda x: x.replace('$', ''))

In [71]:
#convert cost column to numeric
data_original['cost'] = pd.to_numeric(data_original['cost'])

In [72]:
#check that cost is numeric
data_original.dtypes

PMID/PMCID        object
Publisher         object
Journal title     object
Article title     object
cost             float64
j_title           object
                  object
dtype: object

In [73]:
#get mean prices of articles by journal title
data_original.groupby('j_title')['cost'].mean()

j_title
ACADEMY OF NUTRITION AND DIETETICS                                                           2379.540000
ACS CHEMICAL BIOLOGY                                                                         1418.186000
ACS CHEMICAL NEUROSCIENCE                                                                    1186.800000
ACS NANO                                                                                      668.140000
ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL CRYSTALLOGRAPHY                                 771.420000
ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CRYSTALLOGRAPHY                                  773.740000
ACTA CRYSTALLOGRAPHICA SECTION F: STRUCTURAL BIOLOGY AND CRYSTALLIZATION COMMUNICATIONS       796.635000
ACTA CRYSTALLOGRAPHICA, SECTION D                                                             757.180000
ACTA CRYSTALLOGRAPHY D                                                                        774.190000
ACTA D                                         

In [34]:
#get median price of articles by journal title
data_original.groupby('j_title')['cost'].median()

j_title
ACADEMY OF NUTRITION AND DIETETICS                                                           237954.0
ACS CHEMICAL BIOLOGY                                                                         129459.0
ACS CHEMICAL NEUROSCIENCE                                                                    118680.0
ACS NANO                                                                                      66814.0
ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL CRYSTALLOGRAPHY                                 77142.0
ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CRYSTALLOGRAPHY                                  77374.0
ACTA CRYSTALLOGRAPHICA SECTION F: STRUCTURAL BIOLOGY AND CRYSTALLIZATION COMMUNICATIONS       79663.5
ACTA CRYSTALLOGRAPHICA, SECTION D                                                             75718.0
ACTA CRYSTALLOGRAPHY D                                                                        77419.0
ACTA D                                                                    

In [35]:
#get the std dev of articple prices by journal title
data_original.groupby('j_title')['cost'].std()

j_title
ACADEMY OF NUTRITION AND DIETETICS                                                                  NaN
ACS CHEMICAL BIOLOGY                                                                       5.073096e+04
ACS CHEMICAL NEUROSCIENCE                                                                           NaN
ACS NANO                                                                                   3.570889e+03
ACTA CRYSTALLOGRAPHICA SECTION D,  BIOLOGICAL CRYSTALLOGRAPHY                                       NaN
ACTA CRYSTALLOGRAPHICA SECTION D: BIOLOGICAL CRYSTALLOGRAPHY                                        NaN
ACTA CRYSTALLOGRAPHICA SECTION F: STRUCTURAL BIOLOGY AND CRYSTALLIZATION COMMUNICATIONS    1.560585e+03
ACTA CRYSTALLOGRAPHICA, SECTION D                                                                   NaN
ACTA CRYSTALLOGRAPHY D                                                                              NaN
ACTA D                                                  

In [39]:
#example of grouping by two columns
data_original.groupby(['Publisher', 'j_title'])['cost'].mean()

Publisher                                     j_title                                               
ACS                                           BIOMACROMOLECULES                                         2.381040e+05
                                              J MED CHEM                                                6.561000e+04
                                              J ORG CHEM                                                6.858800e+04
                                              JOURNAL OF MEDICINAL CHEMISTRY                            2.392200e+05
                                              JOURNAL OF PROTEOME RESEARCH                              2.367950e+05
                                              MOL PHARM                                                 6.493300e+04
ACS (Amercian Chemical Society) Publications  ACS CHEMICAL BIOLOGY                                      1.294685e+05
                                              JOURNAL OF CHEMICAL INFORMATION AN